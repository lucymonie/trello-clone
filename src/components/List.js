import React from 'react';
import AddCard from './AddCard';
import Card from './Card';

const List = ({
  listTitle,
  listId,
  tasks,
  updateTasks,
  editTask,
  handleTaskText,
  handleDueDate,
  date,
  taskText,
  showEditTask,
  patchTask,
  onClickCancelTask
}) => (
<div className="list-wrapper">
  <div className="list blue-grey lighten-4">
    <div className="list-header"><b>{ listTitle.charAt(0).toUpperCase().concat(listTitle.slice(1)) }</b></div>
    {tasks.map((task, index) => (
      <Card
        key={index}
        task={task}
        editTask={editTask}
        handleTaskText={handleTaskText}
        handleDueDate={handleDueDate}
        date={date}
        taskText={taskText}
        showEditTask={showEditTask}
        patchTask={patchTask}
        onClickCancelTask={onClickCancelTask}
      />
    ))}
    <AddCard listId={listId} updateTasks={updateTasks} />
  </div>
</div>
)

export default List;
