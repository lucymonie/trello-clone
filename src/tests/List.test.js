import React from 'react';
import { shallow } from 'enzyme';
import List from '../components/List';


describe("List", () => {
  it("renders consistently", () => {
    const component = shallow(<List updateTasks={jest.fn()} listTitle="" tasks={[]} />);
    expect(component).toMatchSnapshot();
  });

  it("should render the capitalised list title that is passed in", () => {
    const component = shallow(<List updateTasks={jest.fn()} listTitle="learning" tasks={[]} />);
    const titleDiv = component.find("div").at(2);
    expect(titleDiv.text()).toBe("Learning");
  });

});
