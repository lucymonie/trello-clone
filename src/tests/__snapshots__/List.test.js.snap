// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List renders consistently 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <List
    listTitle=""
    tasks={Array []}
    updateTasks={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="list indigo lighten-5"
>
        <div
                className="list-header"
        >
                <b />
        </div>
        <AddCard
                listId={undefined}
                updateTasks={[Function]}
        />
</div>,
      "className": "list-wrapper",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="list-header"
>
            <b />
</div>,
          Array [],
          <AddCard
            listId={undefined}
            updateTasks={[Function]}
/>,
        ],
        "className": "list indigo lighten-5",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <b />,
            "className": "list-header",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "",
            },
            "ref": null,
            "rendered": "",
            "type": "b",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "listId": undefined,
            "updateTasks": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="list indigo lighten-5"
>
          <div
                    className="list-header"
          >
                    <b />
          </div>
          <AddCard
                    listId={undefined}
                    updateTasks={[Function]}
          />
</div>,
        "className": "list-wrapper",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="list-header"
>
              <b />
</div>,
            Array [],
            <AddCard
              listId={undefined}
              updateTasks={[Function]}
/>,
          ],
          "className": "list indigo lighten-5",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <b />,
              "className": "list-header",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "",
              },
              "ref": null,
              "rendered": "",
              "type": "b",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "listId": undefined,
              "updateTasks": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
